sszgen/unmarshal.go

```diff
@@ -34,7 +34,7 @@ func (v *Value) unmarshal(dst string) string {
 			return fmt.Sprintf("copy(::.%s[:], %s)", v.name, dst)
 		}
 		validate := ""
-		if v.s == 0 {
+		if !v.isFixed() {
```

Comment:
Some of the codegen methods relied on callers setting s equal to 0 as a way of signaling downstream. This PR removes all instances of that pattern and replaces with explicit runtime checks based on parsed structure.